## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 452993`
Commit: `65191d62ddbdb7942b0948d7937026ea4ab886f9`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>libubox-mbedtls</li>
    <li>ustream-ssl-mbedtls</li>
  </ul>
</details>
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>netifd</li>
    <li>ubus</li>
    <li>uci</li>
    <li>uclient</li>
    <li>udebug</li>
    <li>uqmi</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>libubox</li>
    <li>libubox-nossl</li>
    <li>libubox-wolfssl</li>
    <li>ustream-ssl</li>
    <li>ustream-ssl-wolfssl</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>ubus</summary>
<pre>Running phase: configurePhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;configurePhase&quot;}
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/share/doc/ubus -DCMAKE_INSTALL_INFODIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/include -DCMAKE_INSTALL_SBINDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/4m39xvj0wj6xiz6l6ryvmfizm13d2xan-ubus-unstable-2023-12-18 -DBUILD_LUA=OFF
-- The C compiler identification is GNU 14.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at lua/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake &lt; 3.5 has been removed from CMake.

  Update the VERSION argument &lt;min&gt; value.  Or, use the &lt;min&gt;...&lt;max&gt; syntax
  to tell CMake that the project requires at least &lt;min&gt; but has been updated
  to work with policies introduced by &lt;max&gt; or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>uci</summary>
<pre>unpacking source archive /nix/store/b1ps5sk3dw18ndxvlpjzfy1ndx3ibjr3-uci-5781664
source root is uci-5781664
Running phase: patchPhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;patchPhase&quot;}
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;updateAutotoolsGnuConfigScriptsPhase&quot;}
Running phase: configurePhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;configurePhase&quot;}
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/share/doc/uci -DCMAKE_INSTALL_INFODIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/include -DCMAKE_INSTALL_SBINDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/5ym42s243c7l1d4prykw0l8hnbqrqll1-uci-unstable-2023-08-10 -DBUILD_LUA=OFF
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake &lt; 3.5 has been removed from CMake.

  Update the VERSION argument &lt;min&gt; value.  Or, use the &lt;min&gt;...&lt;max&gt; syntax
  to tell CMake that the project requires at least &lt;min&gt; but has been updated
  to work with policies introduced by &lt;max&gt; or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>uclient</summary>
<pre>unpacking source archive /nix/store/mis7ngv4zc56k264lhxxl5g9gl26xrss-uclient-007d945
source root is uclient-007d945
Running phase: patchPhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;patchPhase&quot;}
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;updateAutotoolsGnuConfigScriptsPhase&quot;}
Running phase: configurePhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;configurePhase&quot;}
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/share/doc/uclient -DCMAKE_INSTALL_INFODIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/include -DCMAKE_INSTALL_SBINDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/a3nvgp6xyw02b5rsw4d6xwr58q6x9bq8-uclient-unstable-2023-04-13
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake &lt; 3.5 has been removed from CMake.

  Update the VERSION argument &lt;min&gt; value.  Or, use the &lt;min&gt;...&lt;max&gt; syntax
  to tell CMake that the project requires at least &lt;min&gt; but has been updated
  to work with policies introduced by &lt;max&gt; or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>uqmi</summary>
<pre>patching script interpreter paths in .
./data/gen-code.pl: interpreter directive changed from &quot;#!/usr/bin/env perl&quot; to &quot;/nix/store/xi08qryndrv2a2vih9s4n5kzfp8hzn9r-perl-5.40.0/bin/perl&quot;
./data/gen-header.pl: interpreter directive changed from &quot;#! perl&quot; to &quot;/nix/store/xi08qryndrv2a2vih9s4n5kzfp8hzn9r-perl-5.40.0/bin/perl&quot;
./data/gen-error-list.pl: interpreter directive changed from &quot;#!/usr/bin/env perl&quot; to &quot;/nix/store/xi08qryndrv2a2vih9s4n5kzfp8hzn9r-perl-5.40.0/bin/perl&quot;
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;updateAutotoolsGnuConfigScriptsPhase&quot;}
Running phase: configurePhase
@nix {&quot;action&quot;:&quot;setPhase&quot;,&quot;phase&quot;:&quot;configurePhase&quot;}
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/share/doc/uqmi -DCMAKE_INSTALL_INFODIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/include -DCMAKE_INSTALL_SBINDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/frxwlync78r3l94863r2cbnz7ngy273j-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/952jxwkxrvcsmlrqbs2zfd9r17b9c9gk-uqmi-unstable-2024-01-16
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake &lt; 3.5 has been removed from CMake.

  Update the VERSION argument &lt;min&gt; value.  Or, use the &lt;min&gt;...&lt;max&gt; syntax
  to tell CMake that the project requires at least &lt;min&gt; but has been updated
  to work with policies introduced by &lt;max&gt; or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

--- a/pkgs/by-name/ub/ubus/package.nix
+++ b/pkgs/by-name/ub/ubus/package.nix
@@ -9,12 +9,12 @@
 
 stdenv.mkDerivation {
   pname = "ubus";
-  version = "unstable-2023-12-18";
+  version = "unstable-202-10-17";
 
   src = fetchgit {
     url = "https://git.openwrt.org/project/ubus.git";
-    rev = "65bb027054def3b94a977229fd6ad62ddd32345b";
-    hash = "sha256-n82Ub0IiuvWbnlDCoN+0hjo/1PbplEbc56kuOYMrHxQ=";
+    rev = "60e04048a0e2f3e33651c19e62861b41be4c290f";
+    hash = "sha256-fjxO77z+do5gZ7nLwHbC14UnP9cmZ5eANNn4X6Sudn0=";
   };
 
   cmakeFlags = [ "-DBUILD_LUA=OFF" ];

--- a/pkgs/by-name/uc/uci/package.nix
+++ b/pkgs/by-name/uc/uci/package.nix
@@ -9,12 +9,12 @@
 
 stdenv.mkDerivation {
   pname = "uci";
-  version = "unstable-2023-08-10";
+  version = "unstable-2025-10-12";
 
   src = fetchgit {
     url = "https://git.openwrt.org/project/uci.git";
-    rev = "5781664d5087ccc4b5ab58505883231212dbedbc";
-    hash = "sha256-8MyFaZdAMh5oMPO/5QyNT+Or57eBL3mamJLblGGoF9g=";
+    rev = "57c1e8cd2c051d755ca861a9ab38a8049d2e3f95";
+    hash = "sha256-/Ian7WoBvm9nmniHdVTEIyRW1BPTmOe3O0v59aDaXc0=";
   };
 
   hardeningDisable = [ "all" ];

--- a/pkgs/by-name/uc/uclient/package.nix
+++ b/pkgs/by-name/uc/uclient/package.nix
@@ -5,23 +5,29 @@
   cmake,
   pkg-config,
   libubox,
+  ucode,
+  json_c,
 }:
 
 stdenv.mkDerivation {
   pname = "uclient";
-  version = "unstable-2023-04-13";
+  version = "unstable-2025-10-03";
 
   src = fetchgit {
     url = "https://git.openwrt.org/project/uclient.git";
-    rev = "007d945467499f43656b141171d31f5643b83a6c";
-    hash = "sha256-A47dyVc2MtOL6aImZ0b3SMWH2vzjfAXzRAOF4nfH6S0=";
+    rev = "dc909ca71bc884c0e5362e1d7cc7808696cb2add";
+    hash = "sha256-jrhLBB3Mb7FvxMtKxG7e7D/hcyygTjx868POGtF+Dcc=";
   };
 
   nativeBuildInputs = [
     cmake
     pkg-config
   ];
-  buidInputs = [ libubox ];
+  buildInputs = [
+    libubox
+    ucode
+    json_c
+  ];
 
   preConfigure = ''
     sed -e 's|ubox_include_dir libubox/ustream-ssl.h|ubox_include_dir libubox/ustream-ssl.h HINTS ${libubox}/include|g' \

